web/src/main/java/org/springframework/security/web/header/HeaderWriterFilter.java

```diff
@@ -38,55 +39,55 @@
  *
  * @author Marten Deinum
  * @author Josh Cummings
+ * @author Ankur Pathak
  * @since 3.2
- *
  */
 public class HeaderWriterFilter extends OncePerRequestFilter {
 
+
 	/**
-	 * Collection of {@link HeaderWriter} instances to write out the headers to the
+	 * Composite{@link CompositeHeaderWriter} containing Collection of {@link HeaderWriter} instances to write out the headers to the
 	 * response.
 	 */
-	private final List<HeaderWriter> headerWriters;
+	private final CompositeHeaderWriter compositeHeaderWriter;
 
 	/**
 	 * Creates a new instance.
 	 *
 	 * @param headerWriters the {@link HeaderWriter} instances to write out headers to the
-	 * {@link HttpServletResponse}.
+	 *                      {@link HttpServletResponse}.
 	 */
 	public HeaderWriterFilter(List<HeaderWriter> headerWriters) {
 		Assert.notEmpty(headerWriters, "headerWriters cannot be null or empty");
-		this.headerWriters = headerWriters;
+		compositeHeaderWriter = new CompositeHeaderWriter(headerWriters);
 	}
 
 	@Override
 	protected void doFilterInternal(HttpServletRequest request,
-			HttpServletResponse response, FilterChain filterChain)
-					throws ServletException, IOException {
+									HttpServletResponse response, FilterChain filterChain)
+			throws ServletException, IOException {
 
 		HeaderWriterResponse headerWriterResponse = new HeaderWriterResponse(request,
-				response, this.headerWriters);
+				response, compositeHeaderWriter);
 		HeaderWriterRequest headerWriterRequest = new HeaderWriterRequest(request,
 				headerWriterResponse);
 
 		try {
 			filterChain.doFilter(headerWriterRequest, headerWriterResponse);
-		}
-		finally {
+		} finally {
 			headerWriterResponse.writeHeaders();
 		}
 	}
 
 	static class HeaderWriterResponse extends OnCommittedResponseWrapper {
 		private final HttpServletRequest request;
-		private final List<HeaderWriter> headerWriters;
+		private final CompositeHeaderWriter compositeHeaderWriter;
 
 		HeaderWriterResponse(HttpServletRequest request, HttpServletResponse response,
-				List<HeaderWriter> headerWriters) {
+								CompositeHeaderWriter compositeHeaderWriter) {
```

Comment:
Would you please do two indents for continuations? Even with method parameters, this is the formatting convention.